### PR TITLE
[portinglayer] Matter data model

### PR DESCRIPTION
**Pull request by step0035**
- custom data model for ameba series
- able to support dynamic endpoints
- add device type endpoint presets and cluster presets
- light_dm example that uses this ameba data model

### CODE REVIEW

File: component/common/application/matter/core/matter_data_model.h [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+
+typedef std::variant<uint8_t, uint8_t*, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, int64_t, float> AttributeValue;
+
+// Configurations
+struct AttributeConfig
+{
+    std::uint32_t attributeId;
+    std::uint8_t dataType; /* EmberAfAttributeType in string format */
+    EmberAfDefaultOrMinMaxAttributeValue value;
+    std::uint16_t size; /* attributeSize, use ZAP_TYPE(type) */
```
Comment (shuqunx) [inline component/common/application/matter/core/matter_data_model.h:27]: Wrong comment here?

File: component/common/application/matter/example/light_dm/example_matter_light.h [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#ifndef EXAMPLE_MATTER_LIGHT_H
+#define EXAMPLE_MATTER_LIGHT_H
+
+void example_matter_task(void);
```
Comment (xshuqun) [inline component/common/application/matter/example/light_dm/example_matter_light.h:4]: Please make changes here

File: component/common/application/matter/core/matter_data_model.h (hunk 1 above)
Comment (step0035) [reply component/common/application/matter/core/matter_data_model.h]: fixed

File: component/common/application/matter/example/light_dm/example_matter_light.h (hunk 2 above)
Comment (step0035) [reply component/common/application/matter/example/light_dm/example_matter_light.h]: fixed

File: component/common/application/matter/core/matter_data_model.cpp [hunk 3]
```diff
@@ -0,0 +1,978 @@
...
+                endpoints[i].setParentEndpointId(0xFFFF);
+            }
+        }
+    }
+    endpointCount--;
+}
+
+void Node::enableAllEndpoints()
+{
+    for (Endpoint & endpoint: endpoints)
```
Comment (SirRu24) [inline component/common/application/matter/core/matter_data_model.cpp:974]: Line 36, 354, 367, 380, 393, 410, 423, 440, 453, 470, 490, 506, 519, 548, 559, 
562, 567, 572, 584, 591, 605, 862, 871, 887, 896, 899, 904, 909, 921, 936, 974: 
update reference declaration to Attribute &attribute format

**Review by SirRu24 — COMMENTED** [review]
matter_data_model.cpp, example_matter_light.cpp, matter_drivers.cpp, matter_drivers.h: Can change the style of pointer and reference declaration, not urgent
lib_chip_light_core.mk, lib_chip_light_main.mk: Clean the commented out line, I am not sure if it was intended to be left here or forget to delete

**Pull request merged**